internal/cmd/operator-sdk/pkgmantobundle/cmd.go

```diff
@@ -0,0 +1,318 @@
+					return err
+				}
+			}
+
+		}
+	}
+	return nil
+}
+
+func getSDKStampsAndChannels(path string, channelsByCSV map[string][]string) (map[string]string, string, error) {
+	bundle, err := apimanifests.GetBundleFromDir(path)
+	if err != nil {
+		return nil, "", err
+	}
+
+	sdkLabels, err := getSDKStamps(bundle)
+	if err != nil {
+		return nil, "", err
+	}
+
+	// Find channels matching the CSV names
+	channels := getChannelsByCSV(bundle, channelsByCSV)
+
+	return sdkLabels, channels, nil
+}
+
+// getSDKStamps parses the CSV and gets SDK stamps.
+func getSDKStamps(bundle *apimanifests.Bundle) (map[string]string, error) {
+	if bundle.CSV == nil {
+		return nil, fmt.Errorf("cannot find CSV from manifests package")
+	}
+
+	// Extract SDK layout and version from CSV annotations.
+	csvAnnotations := bundle.CSV.GetAnnotations()
+	sdkLabels := make(map[string]string)
+
+	for key, value := range csvAnnotations {
+		if key == metrics.BuilderObjectAnnotation {
+			sdkLabels[key] = value
+		}
+
+		if key == metrics.LayoutObjectAnnotation {
+			sdkLabels[key] = value
+		}
+	}
+
+	return sdkLabels, nil
+}
+
+// getChannelsByCSV creates a list for channels for the currentCSV,
+func getChannelsByCSV(bundle *apimanifests.Bundle, channelsByCSV map[string][]string) (channels string) {
+	// Find channels matching the CSV names
+	var channelNames []string
+	for csv, ch := range channelsByCSV {
+		if csv == bundle.CSV.GetName() {
+			channelNames = ch
+			break
+		}
+	}
+	channels = strings.Join(channelNames, ",")
+
+	// TODO: verify if we have to add this validation since while building bundles if channel is not specified
+	// we add the default channel.
+	if channels == "" {
+		channels = "preview"
+		log.Infof("Supported channels cannot be identified from CSV %s, using default channel 'preview'", bundle.CSV.GetName())
+	}
+
+	return channels
+}
+
+func getPackageMetadata(pkg *apimanifests.PackageManifest) (packagename, defaultChannel string, channelsByCSV map[string][]string, err error) {
+	packagename = pkg.PackageName
+	if packagename == "" {
+		err = fmt.Errorf("cannot find packagename from the manifest directory")
+		return
+	}
+
+	defaultChannel = pkg.DefaultChannelName
+	if defaultChannel == "" {
+		defaultChannel = "preview"
+	}
+
+	channelsByCSV = make(map[string][]string)
+
+	for _, p := range pkg.Channels {
+		if _, ok := channelsByCSV[p.CurrentCSVName]; !ok {
+			channelsByCSV[p.CurrentCSVName] = make([]string, 0)
+		}
+		channelsByCSV[p.CurrentCSVName] = append(channelsByCSV[p.CurrentCSVName], p.Name)
+	}
+
+	return
+}
+
+func (p *pkgManToBundleCmd) validate(args []string) error {
+	if len(args) != 1 {
+		return fmt.Errorf("a package manifest directory argument is required")
+	}
+
+	if len(p.baseImg) == 0 && len(p.buildCmd) != 0 {
+		return fmt.Errorf("base image needs to be specified to build bundle image")
+	}
+	return nil
+}
+
+func (p *pkgManToBundleCmd) setDefaults() {
+	if p.outputDir == "" {
+		p.outputDir = "bundle"
+	}
```

Comment:
Why don't we set the default in the CLI? Also I think this needs to be bundles. based on Eric's comment to the EP. Maybe I need to address those in the EP :( 
https://github.com/operator-framework/enhancements/pull/64/files#r597990671

```
 	pkgManToBundleCmd.Flags().StringVar(&p.outputDir, "output-dir", "bundles", "Directory to write bundle to.")
```